python/location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,262 @@
+# This Python file is an extended/modified version of Jerry Wu's audio_prep.py
+# This python file is a script to download a given label from AudioSet
+# The downloaded audios would be clipped into 10 seconds periods
+# The downloaded audio segments are then processed using Librosa to extract features
+# Stored in eval_segments, balanced_train, and unbalanced_train separately.
+
+# Usage: python3 audio_prep.py <label>
+
+
+from __future__ import unicode_literals
+import youtube_dl
+import csv
+import json
+from pydub import AudioSegment
+import os
+import shutil
+from os import listdir
+from os.path import isfile, join
+import sys
+import numpy as np
+import librosa
+import argparse
+import numpy as np
+
+file_structure = "python/location/lbs/activity/audioset/dataprocessing/"
+referDoc = "ReferDoc/"
+downloaded_audio = []
+feature_dict = {}
+
+
+def setup_commandline_args():
+  parser = argparse.ArgumentParser(description='Process dataset.csv to input into TensorFlow.')
+  parser.add_argument('-d', '--datasets', required=True, type=str, nargs='+')
+  parser.add_argument('-r', '--redo', action='store_true', required=False)
+  parser.add_argument('-l', '--labels', required=True, type=str, nargs='+')
+  args = parser.parse_args()
+  print(args)
+  return args
+
+
+def parse_csv(csv_file):
+  # Extract all labeled audio info from csv
+  # input: str csv_file
+  # output: list audio_list
+  audio_list = []
+  with open(csv_file, newline='') as csvfile:
+    info = csv.reader(csvfile, delimiter=' ', quotechar='|')
+    for row in info:
+      r = ','.join(row)
+      if '#' not in r:
+        r_lst = r.split(',')
+        url, start_time, end_time = r_lst[0], float(r_lst[2]), float(r_lst[4])
```

Comment:
I think this will be better off stored in a dictionary so that we'll be able to use indices like audio_list[0]["start_time"] for developer readability.